src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -134,13 +144,46 @@ export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
     };
   }, [relatedVenueEvents, userEventIds, selectedDayIndex, getEventLocation]);
 
+  const downloadPersonalEventsCalendar = useCallback(() => {
+    const dayStart = addDays(startOfToday(), selectedDayIndex);
+    const allPersonalEvents: PersonalizedVenueEvent[] = relatedVenueEvents
+      .map(
+        prepareForSchedule({
+          day: dayStart,
+          usersEvents: userEventIds,
+          isForCalendarFile: true,
+        })
+      )
+      .filter((event) => event.isSaved);
```

Comment:
I'm guessing `.isSaved` only becomes available here after the `.map`? If so, all good. If not, we should move the `.filter` before the `.map`